src/sql/src/plan/statement/ddl.rs

```diff
@@ -1407,7 +1411,217 @@ pub fn plan_create_views(
+                                                &t.namespace,
+                                                &t.name
+                                            ])
+                                        ),
+                                    ]
+                                }));
+                            let wanted: HashSet<String> =
+                                HashSet::from_iter(targets.iter().map(|t| t.name.to_string()));
+                            let diff = wanted.difference(&known);
+                            if diff.clone().count() > 0 {
+                                error!("During create views user specified tables not found: {:?}  Known tables: {:?}", diff, known);
+                                return Err(anyhow!(
+                                    "Table(s) {} not found in source, try recreating materialized source.",
+                                    diff.map(|d| d.to_owned())
+                                        .collect::<Vec<String>>()
+                                        .join(", ")
+                                ));
+                            }
+                            targets
+                        }
+                        None => details
+                            .tables
+                            .iter()
+                            .map(|t| {
+                                let name =
+                                    UnresolvedObjectName::qualified(&[&t.namespace, &t.name]);
+                                CreateViewsSourceTarget {
+                                    name: name.clone(),
+                                    alias: Some(name),
+                                }
+                            })
+                            .collect(),
+                    };
+
+                    let mut viewdefs = Vec::with_capacity(details.tables.len());
+
+                    // An index from table_name -> schema_name -> PostgresTable
+                    let mut details_info_idx: HashMap<String, HashMap<String, PostgresTable>> =
+                        HashMap::new();
+                    for table in details.tables {
+                        details_info_idx
+                            .entry(table.name.clone())
+                            .or_default()
+                            .entry(table.namespace.clone())
+                            .or_insert(table);
+                    }
+
+                    for target in targets {
+                        let view_name = target.alias.clone().unwrap_or_else(|| target.name.clone());
+                        let name = normalize::unresolved_object_name(target.name.clone())
+                            .map_err(anyhow::Error::new)?;
+                        let schemas = details_info_idx.get(&name.item).ok_or_else(|| {
+                            anyhow!("table {} not found in upstream database", name)
+                        })?;
+                        let t = if let Some(schema) = &name.schema {
```

Comment:
since you're handling the else case of an `if let Some(..)` statement it's more idiomatic to use a `match` statement